src/System.Text.Json/tests/NewtonTests/EnumConverterTests.cs

```diff
@@ -0,0 +1,532 @@
+        {
+            JsonException ex = Assert.Throws<JsonException>(() =>
+            {
+                JsonSerializer.Parse<StoreColor>("\"-1\"");
+            });
+        }
+
+        [Fact]
+        public static void InvalidValueDash()
+        {
+            JsonException ex = Assert.Throws<JsonException>(() =>
+            {
+                JsonSerializer.Parse<StoreColor>("\"-\"");
+            });
+        }
+
+        public enum NegativeEnum
+        {
+            Negative = -1,
+            Zero = 0,
+            Positive = 1
+        }
+
+        [Flags]
+        public enum NegativeFlagsEnum
+        {
+            NegativeFour = -4,
+            NegativeTwo = -2,
+            NegativeOne = -1,
+            Zero = 0,
+            One = 1,
+            Two = 2,
+            Four = 4
+        }
+
+        public class NegativeEnumClass
+        {
+            public NegativeEnum Value1 { get; set; }
+            public NegativeEnum Value2 { get; set; }
+        }
+
+        public class NegativeFlagsEnumClass
+        {
+            public NegativeFlagsEnum Value1 { get; set; }
+            public NegativeFlagsEnum Value2 { get; set; }
+        }
+
+        [Fact]
+        public static void DeserializeNegativeEnum()
+        {
+            string json = @"{
+  ""Value1"": -1,
+  ""Value2"": -2147483648
+}";
+            NegativeEnumClass negativeEnumClass = JsonSerializer.Parse<NegativeEnumClass>(json);
+            Assert.Equal(NegativeEnum.Negative, negativeEnumClass.Value1);
+            Assert.Equal((NegativeEnum)int.MinValue, negativeEnumClass.Value2);
+        }
+
+        [Fact]
+        public static void DeserializeEnumClass()
+        {
+            string json = @"{
+  ""StoreColor"": 2,
+  ""NullableStoreColor1"": 8,
+  ""NullableStoreColor2"": null
+}";
+            EnumClass enumClass = JsonSerializer.Parse<EnumClass>(json);
+            Assert.Equal(StoreColor.Red, enumClass.StoreColor);
+            Assert.Equal(StoreColor.White, enumClass.NullableStoreColor1);
+            Assert.Equal(null, enumClass.NullableStoreColor2);
+        }
+
+        [Fact]
+        public static void DeserializeFlagEnum()
+        {
+            string json = @"{
+  ""StoreColor"": 10,
+  ""NullableStoreColor1"": 0,
+  ""NullableStoreColor2"": 11
+}";
+            EnumClass enumClass = JsonSerializer.Parse<EnumClass>(json);
+            Assert.Equal(StoreColor.Red | StoreColor.White, enumClass.StoreColor);
+            Assert.Equal((StoreColor)0, enumClass.NullableStoreColor1);
+            Assert.Equal(StoreColor.Red | StoreColor.White | StoreColor.Black, enumClass.NullableStoreColor2);
+        }
+
+        [Flags]
+        public enum StoreColor
+        {
+            Black = 1,
+            Red = 2,
+            Yellow = 4,
+            White = 8,
+            DarkGoldenrod = 16
+        }
+    }
+
+    [DataContract]
+    public class DuplicateEnumNameTestClass
+    {
+        [DataMember]
```

Comment:
Same here/elsewhere. I would remove all these attributes unless they were actually needed for some test scenario.